src/Illuminate/View/Compilers/ComponentTagCompiler.php

```diff
@@ -153,7 +153,19 @@ protected function componentString(string $component, array $attributes)
 
         [$data, $attributes] = $this->partitionDataAndAttributes($class, $attributes);
 
-        return " @component('{$class}', [".$this->attributesToString($data->all()).'])
+        // If the component doesn't exists as a class we'll assume it's a class-less component
+        // and pass the component as a view parameter to the data collection bag.
+        if (! class_exists($class)) {
+            $data->put('view', "'$class'");
+
+            $class = ClassLessComponent::class;
+
+            $parameters = "'data' => [".$this->attributesToString($data->all()).']';
+        } else {
+            $parameters = $this->attributesToString($data->all());
+        }
+
+        return " @component('{$class}', [".$parameters.'])
 <?php $component->withAttributes(['.$this->attributesToString($attributes->all()).']); ?>';
```

Comment:
I wanted to use `var_export` for this instead but unfortunally the blade compiler doesn't plays well with the old array syntax.

Reply:
You can get short array syntax by pretty printing using php-parser, I think?

Reply:
Maybe overkill. Feel free to pr afterwards if you want.

Reply:
Yeh, definitely overkill. We can mark this as resolved.